src/action/client_side_file_finder/expand_groups.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+use lazy_static::lazy_static;
+use regex::{Regex};
+
+/// Performs group expansion on a given path.
+/// For example, given path `foo/{bar,baz}/{quux,norf}` this method will yield
+///   `foo/bar/quux`, `foo/bar/norf`, `foo/baz/quux`, `foo/baz/norf`.
+/// The order of expansion is defined as if there were generated by `for` loops nested in
+/// order of groups in the path.
+///
+/// Differences from GRR:
+/// - support for empty value in the 1st and 2nd alternative e.g.:
+///   - `/{,asd}`, `/{asd,}` are resolved here, but not in GRR
+///   - `/{a,b,}` are resolved both in GRR and in here
+/// - support for a single alternative e.g.:
+///   - `/{asd}` is resolved here, but not in GRR
+pub fn expand_groups(path: &str) -> Vec<String> {
+    lazy_static! {
+        static ref RE : Regex = Regex::new(r"\{[^\{]*?\}").unwrap();
+    }
+
+    let mut offset = 0;
+    let mut chunks : Vec<Vec<String>> = vec![];
+
+    for m in RE.find_iter(path){
+        chunks.push(vec![path[offset..m.start()].to_owned()]);
+        chunks.push(path[m.start()+1..m.end()-1].split(',').map(|x| x.to_owned()).collect());
```

Comment:
There is *a lot* of allocations flying around. Are you sure it cannot be avoided?

Reply:
Removed the allocations by changing the type of `chunks` to `Vec<Vec<&str>>`.